src/backend/distributed/master/master_modify_multiple_shards.c

```diff
@@ -241,10 +275,19 @@ SendQueryToShards(Query *query, List *shardIntervalList, Oid relationId)
 											   &shardConnectionsFound);
 		Assert(shardConnectionsFound);
 
-		deparse_shard_query(query, relationId, shardId, shardQueryString);
+		if (truncateCommand)
+		{
+			deparse_truncate_query(query, relationId, shardId, shardQueryString);
+		}
+		else
+		{
+			deparse_shard_query(query, relationId, shardId, shardQueryString);
```

Comment:
I am hesitant to touch ruleutils file as they need to be kept in sync for each postgresql version.